src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -43,26 +43,44 @@ public override void Initialize (AnalysisContext context)
 				context.RegisterSymbolAction (symbolAnalysisContext => {
 					var methodSymbol = (IMethodSymbol) symbolAnalysisContext.Symbol;
 					CheckMatchingAttributesInOverrides (symbolAnalysisContext, methodSymbol);
+					CheckAttributeCtor (symbolAnalysisContext, methodSymbol);
+					foreach (var typeParameter in methodSymbol.TypeParameters)
+						CheckAttributeCtor (symbolAnalysisContext, typeParameter);
+
 				}, SymbolKind.Method);
 
 				context.RegisterSymbolAction (symbolAnalysisContext => {
 					var typeSymbol = (INamedTypeSymbol) symbolAnalysisContext.Symbol;
 					CheckMatchingAttributesInInterfaces (symbolAnalysisContext, typeSymbol);
+					CheckAttributeCtor (symbolAnalysisContext, typeSymbol);
+					foreach (var typeParameter in typeSymbol.TypeParameters)
+						CheckAttributeCtor (symbolAnalysisContext, typeParameter);
+
 				}, SymbolKind.NamedType);
 
-				if (AnalyzerDiagnosticTargets.HasFlag (DiagnosticTargets.Property)) {
-					context.RegisterSymbolAction (symbolAnalysisContext => {
-						var propertySymbol = (IPropertySymbol) symbolAnalysisContext.Symbol;
+
+				context.RegisterSymbolAction (symbolAnalysisContext => {
+					var propertySymbol = (IPropertySymbol) symbolAnalysisContext.Symbol;
+					if (AnalyzerDiagnosticTargets.HasFlag (DiagnosticTargets.Property)) {
 						CheckMatchingAttributesInOverrides (symbolAnalysisContext, propertySymbol);
-					}, SymbolKind.Property);
-				}
+					}
+
+					CheckAttributeCtor (symbolAnalysisContext, propertySymbol);
```

Comment:
Don't we need to do this for getter/setter as well (similar to what we do for events with add/remove).

Reply:
In fact, RUC is not allowed on the property itself, only on the methods...

Reply:
This is needed for `RequiresAssemblyFilesAttribute`, which can be put on properties.